crates/vfio-ioctls/src/vfio_device.rs

```diff
@@ -166,9 +73,10 @@ struct vfio_region_info_with_cap {
 ///
 /// Multiple VFIO groups may be associated with the same VFIO container to share the underline
 /// address translation mapping tables.
+#[allow(dead_code)]
```

Comment:
Why do we need to allow dead_code?

Reply:
When neither kvm nor mshv is enabled, the field `VfioContainer::device_fd` is not used. I  could move "#[allow(dead_code)] to the file `device_fd`.

Reply:
Add comments:
```
// Safety:
// - absolutely trust the underlying kernel
// - absolutely trust data returned by the underlying kernel
// - assume kernel will return error if caller passes in invalid file handle, parameter or buffer.
```

Reply:
Why not have `device_fd` hidden behind a feature instead?

Reply:
The PR #33 goes that way, but the conditional compilation is complex, so I hope to unify the interface.